pkg/block/namespace.go

```diff
@@ -89,10 +89,9 @@ func GetStorageType(namespaceURL *url.URL) (StorageType, error) {
 }
 
 func formatPathWithNamespace(namespacePath, keyPath string) string {
-	namespacePath = strings.TrimSuffix(namespacePath, "/")
-	keyPath = strings.TrimPrefix(keyPath, "/")
+	namespacePath = strings.Trim(namespacePath, "/")
 	if len(namespacePath) == 0 {
-		return keyPath
+		return strings.TrimPrefix(keyPath, "/")
 	}
 	return namespacePath + "/" + keyPath
```

Comment:
Where is the // bug in the previous implementation? it makes sure that there will be no '/' before that path and just one '/' when concatenating the storage namespace.

In the new implementation if keyPath will have a '/' prefix it will contact '//' in case the namespace is not empty.

Reply:
> In the new implementation if keyPath will have a '/' prefix it will contact '//' in case the namespace is not empty.

Which is right. If a key starts with a slash, you expect to see it as part of the path. However, if it doesn't, you don't expect an additional slash to be added from the `namespacePath`.

Look at the tests, I think the examples there explains it the best.

Reply:
Oh and the `namespacePath` starts with a slash when it's not empty, therefore you need to trim both prefix and suffix.